### DISCUSSION TITLE
Why is it not possible to add a request value of type object.

### DISCUSSION THREAD

**Discussion by Silinator** [laravel/framework#45567] [Q&A]
- Laravel Version: 9.45.1
- PHP Version: 8.2.0
Database Driver & Version: mysql & 10.7.4-MariaDB-1:10.7.4+maria~focal

### Description:
I was trying to add my decoded jwt token to the $request class (inside a middleware)

### Steps To Reproduce:
```php
public function handle(Request $request, Closure $next) {
  $tokenData = (new AuthUserService())->getData($request->accessToken);
  $request['tokenData'] = $tokenData;
  return $next($request);
}
```
Error:
`Expected a scalar, or an array as a 2nd argument to "Symfony\Component\HttpFoundation\InputBag::set()", "stdClass" given.`

Maybe related: https://github.com/symfony/symfony/issues/44432

Not so great workarounds:
```php
$request['tokenData'] = (array) $tokenData;
```

```php
$request['tokenData'] = json_encode($tokenData);
```

**Comment by timacdonald** [comment]
An `InputBag` is meant for storing input from a HTTP request. This means it only supports types that may be sent over HTTP (not rich class based objects).

I would recommend setting this as a request attribute instead:

```php
public function handle(Request $request, Closure $next) {
  $tokenData = (new AuthUserService())->getData($request->accessToken);

  $request->attributes->set('tokenData', $tokenData);

  return $next($request);
}
```

```php
$tokenData = $request->attributes->get('tokenData');
```
